SIPS/sip-32.md

```diff
@@ -0,0 +1,95 @@
+---
+sip: 32
+title: Add `snap_trackEvent` method for pre-installed Snaps
+status: Draft
+author: Daniel Rocha (@danroc)
+created: 2025-03-20
+---
+
+## Abstract
+
+This Snap Improvement Proposal (SIP) introduces a new method,
+`snap_trackEvent`, allowing pre-installed Snaps to submit tracking events
+through the client. The client will determine how to handle these events.
+
+This feature enables Snaps to utilize the existing MetaMask client
+infrastructure for event tracking while maintaining user privacy.
+
+## Motivation
+
+Currently, there is no standardized way for pre-installed Snaps to submit
+analytics or tracking events. This proposal aims to:
+
+- Enable pre-installed Snaps to reuse MetaMask’s event tracking infrastructure.
+
+- Allow the MetaMask client to control event handling, ensuring security and
+  compliance with privacy policies.
+
+## Specification
+
+> The `snap_trackEvent` method allows pre-installed Snaps to submit structured
+> tracking events to the client. It is the client’s responsibility to process
+> these events accordingly.
+>
+> It's the client's responsibility to ensure that only pre-installed Snaps can
+> call this method.
+
+### `snap_trackEvent`
+
+The `snap_trackEvent` method is defined as follows:
+
+#### Parameters
+
+- `event` - Event object to be tracked.
+  - `category: string` - The category to associate the event.
+
+  - `name: string` - The name of the event to track.
```

Comment:
Do we need to uphold any rules for the format for these?

Reply:
I don't think so, at least I didn't find any recommendation about it. I'll try to check with the data team.

Reply:
I got this answer from data:

> The constraints what we impose semantically in order to preserve the continuity of the analytics; same name and same format as the event fired by extension, newly installed snaps, etc.

It seems that we cannot enforce a name format here.